src/Momento.Sdk/Responses/CacheGetResponse.cs

```diff
@@ -1,11 +1,52 @@
 ﻿using Google.Protobuf;
 using Momento.Protos.CacheClient;
-
+using Momento.Sdk.Exceptions;
 namespace Momento.Sdk.Responses;
 
-public class CacheGetResponse : CacheGetResponseBase
+public abstract class CacheGetResponse
 {
-    public CacheGetResponse(_GetResponse response) : base(response.Result, response.CacheBody)
+    public class Hit : CacheGetResponse
+    {
+        public CacheGetStatus Status { get => CacheGetStatus.HIT; }
+        protected readonly ByteString value;
```

Comment:
cool, thanks.  I think we should merge this as-is because it's a good step forward, but would like to circle back quickly and see if the enum is still necessary at all.  It feels like the only thing that is currently forcing us to keep it is that batch api, and if that's true, we should talk about options for getting rid of that bit, because having the enum here feels inconsistent since we got rid of it in the other places.

Reply:
That sounds good. The enum shouldn't be necessary but removing it altogether as part of this effort seemed risky.